cpp/src/arrow/compute/kernels/hash_aggregate.cc

```diff
@@ -541,9 +616,27 @@ struct GrouperFastImpl : Grouper {
   ~GrouperFastImpl() { map_.cleanup(); }
 
   Result<Datum> Consume(const ExecBatch& batch) override {
+    // ARROW-14027: expand scalar arguments for now
```

Comment:
Nit: elsewhere we call "materialization of a batch.length constant array from a scalar" broadcasting, I'd prefer to keep that consistent
```suggestion
    // ARROW-14027: broadcast scalar arguments for now
```